sdks/python/apache_beam/io/avroio_test.py

```diff
@@ -58,424 +63,509 @@
   snappy = None  # pylint: disable=invalid-name
   logging.warning('python-snappy is not installed; some tests will be skipped.')
 
+import sys
 
-class TestAvro(unittest.TestCase):
-
-  _temp_files = []
-
-  def __init__(self, methodName='runTest'):
-    super(TestAvro, self).__init__(methodName)
-    self.use_fastavro = False
-
-  @classmethod
-  def setUpClass(cls):
-    # Method has been renamed in Python 3
-    if sys.version_info[0] < 3:
-      cls.assertCountEqual = cls.assertItemsEqual
-
-  def setUp(self):
-    # Reducing the size of thread pools. Without this test execution may fail in
-    # environments with limited amount of resources.
-    filebasedsource.MAX_NUM_THREADS_FOR_SIZE_ESTIMATION = 2
-
-  def tearDown(self):
-    for path in self._temp_files:
-      if os.path.exists(path):
-        os.remove(path)
-    self._temp_files = []
-
-  RECORDS = [{'name': 'Thomas',
-              'favorite_number': 1,
-              'favorite_color': 'blue'}, {'name': 'Henry',
-                                          'favorite_number': 3,
-                                          'favorite_color': 'green'},
-             {'name': 'Toby',
-              'favorite_number': 7,
-              'favorite_color': 'brown'}, {'name': 'Gordon',
-                                           'favorite_number': 4,
-                                           'favorite_color': 'blue'},
-             {'name': 'Emily',
-              'favorite_number': -1,
-              'favorite_color': 'Red'}, {'name': 'Percy',
-                                         'favorite_number': 6,
-                                         'favorite_color': 'Green'}]
-
-  SCHEMA = Parse('''
-  {"namespace": "example.avro",
-   "type": "record",
-   "name": "User",
-   "fields": [
-       {"name": "name", "type": "string"},
-       {"name": "favorite_number",  "type": ["int", "null"]},
-       {"name": "favorite_color", "type": ["string", "null"]}
-   ]
-  }
-  ''')
-
-  def _write_data(self,
+RECORDS = [
+    {'name': 'Thomas',
+     'favorite_number': 1,
+     'favorite_color': 'blue'},
+    {'name': 'Henry',
+     'favorite_number': 3,
+     'favorite_color': 'green'},
+    {'name': 'Toby',
+     'favorite_number': 7,
+     'favorite_color': 'brown'},
+    {'name': 'Gordon',
+     'favorite_number': 4,
+     'favorite_color': 'blue'},
+    {'name': 'Emily',
+     'favorite_number': -1,
+     'favorite_color': 'Red'},
+    {'name': 'Percy',
+     'favorite_number': 6,
+     'favorite_color': 'Green'}
+]
+
+# Wrapping TestAvroBase to avoid TestAvroBase being used in the unittest
+class TestAvroBaseWrapper:
```

Comment:
Why is this wrapper needed? If it's just to not execute it, we can probably just change the name.
This might also be the reason why git shows such a big diff, which makes it hard to review. Let's try to fix this.